soc/silabs/silabs_siwx917/siwg917/soc.c

```diff
@@ -16,7 +16,8 @@ int silabs_siwx917_init(void)
 
 	/* FIXME: do not hardcode UART instances */
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(ulpuart0), okay)
-	RSI_ULPSS_UlpUartClkConfig(ULPCLK, ENABLE_STATIC_CLK, 0, ULP_UART_ULP_32MHZ_RC_CLK, 1);
+	RSI_PS_UlpssPeriPowerUp(ULPSS_PWRGATE_ULP_UART);
+	RSI_ULPSS_UlpUartClkConfig(ULPCLK, ENABLE_STATIC_CLK, 0, ULP_UART_REF_CLK, 1);
```

Comment:
Another note: in practice `ULP_UART_REF_CLK` is the same as `ULP_UART_ULP_32MHZ_RC_CLK`, however I went with the former as that's what WiseConnect uses as a default for the ULP UART.